debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/tools/ddrinteractive/commands/DumpAllSparseHeapEntriesCommand.java

```diff
@@ -0,0 +1,93 @@
+package com.ibm.j9ddr.vm29.tools.ddrinteractive.commands;
+
+import java.io.PrintStream;
+
+import com.ibm.j9ddr.CorruptDataException;
+import com.ibm.j9ddr.NoSuchFieldException;
+import com.ibm.j9ddr.tools.ddrinteractive.Context;
+import com.ibm.j9ddr.tools.ddrinteractive.DDRInteractiveCommandException;
+import com.ibm.j9ddr.tools.ddrinteractive.Command;
+import com.ibm.j9ddr.vm29.j9.HashTable;
+import com.ibm.j9ddr.vm29.j9.ObjectToSparseDataHashTable;
+import com.ibm.j9ddr.vm29.j9.SlotIterator;
+import com.ibm.j9ddr.vm29.j9.gc.GCBase;
+import com.ibm.j9ddr.vm29.pointer.generated.J9BuildFlags;
+import com.ibm.j9ddr.vm29.pointer.generated.J9HashTablePointer;
+import com.ibm.j9ddr.vm29.pointer.generated.MM_GCExtensionsPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.MM_SparseAddressOrderedFixedSizeDataPoolPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.MM_SparseDataTableEntryPointer;
+import com.ibm.j9ddr.vm29.pointer.generated.MM_SparseVirtualMemoryPointer;
+import com.ibm.j9ddr.vm29.pointer.VoidPointer;
+import com.ibm.j9ddr.vm29.types.UDATA;
+
+
+public class DumpAllSparseHeapEntriesCommand extends Command
+{
+	public DumpAllSparseHeapEntriesCommand()
+	{
+		addCommand("dumpalloffheapentries", "cmd|help", "dump all off heap entries");
+	}
+
+	public void run(String command, String[] args, Context context, PrintStream out) throws DDRInteractiveCommandException
+	{
+		if (0 != args.length) {
+			String argument = args[0];
+
+			if(argument.equalsIgnoreCase("help")) {
+				help(out);
+				return;
+			}
+		}
+
+		try {
+			MM_GCExtensionsPointer extensions = GCBase.getExtensions();
+			MM_SparseVirtualMemoryPointer sparseVirualMemory = extensions.largeObjectVirtualMemory();
+			if (sparseVirualMemory.notNull()) {
+				MM_SparseAddressOrderedFixedSizeDataPoolPointer sparseAddressOrderedFixedSizeDataPool = sparseVirualMemory._sparseDataPool();
+				if (sparseAddressOrderedFixedSizeDataPool.notNull()) {
+					J9HashTablePointer objectToSparseDataTable = sparseAddressOrderedFixedSizeDataPool._objectToSparseDataTable();
+					if (objectToSparseDataTable.notNull()) {
+						try {
+							HashTable<MM_SparseDataTableEntryPointer> readObjectToSparseDataTable = ObjectToSparseDataHashTable.fromJ9HashTable(objectToSparseDataTable);
+							SlotIterator<MM_SparseDataTableEntryPointer> readSlotIterator = readObjectToSparseDataTable.iterator();
+							long count = readObjectToSparseDataTable.getCount();
+							out.printf("List off heap entries(%,d entries)\n", count);
+							if (0 < count) {
+								if (!J9BuildFlags.J9VM_ENV_DATA64) {
+									throw new CorruptDataException("off heap would not be supported in 32 bit platform.");
```

Comment:
Please hyphenate "32-bit", and improve wording:
```java
									throw new CorruptDataException("off heap is not supported on 32-bit platforms");
```

Reply:
On second thought, this command should only be added in `GetCommandsTask` for 64-bit core files.